pydoctor/templatewriter/pages/functionchild.py

```diff
@@ -41,21 +41,20 @@ def functionAnchor(self, request: object, tag: Tag) -> "Flattenable":
     def shortFunctionAnchor(self, request: object, tag: Tag) -> "Flattenable":
         return self.ob.name
 
+    @renderer
+    def overloads(self, request: object, tag: Tag) -> "Flattenable":
+        return list(format_overloads(self.ob))
+
     @renderer
     def decorator(self, request: object, tag: Tag) -> "Flattenable":
         return list(format_decorators(self.ob))
 
     @renderer
     def functionDef(self, request: object, tag: Tag) -> "Flattenable":
-        def_stmt = 'async def' if self.ob.is_async else 'def'
-        name = self.ob.name
-        if name.endswith('.setter') or name.endswith('.deleter'):
-            name = name[:name.rindex('.')]
-        return [
-            tags.span(def_stmt, class_='py-keyword'), ' ',
-            tags.span(name, class_='py-defname'), 
-            format_signature(self.ob), ':'
-            ]
+        # Exclude function definition if there are overloads
+        if self.ob.overloads:
+            return ()
```

Comment:
Isn't that line supposed to be tested?

Reply:
I will add an assertion or test that confirms that the main signature is excluded from output

Reply:
Moved to the format function and tested